plasmapy/utils/data/downloader.py

```diff
@@ -4,74 +4,409 @@
+            with keys `sha` and `download_url`.
+        """
+        # If the current blob file has been updated in the past 5 minutes,
+        # don't bother doing it again
+        # Ignore in tests, as this won't happen in CI
+        try:
+            if time.time() - self._blob_dict["_timestamp"] < 300:
+                return None  # coverage : ignore
+        # If the _timestamp key hasn't been set yet, the blob file has
+        # never been updated before
+        except KeyError:  # coverage : ignore
+            pass
+
+        # If this instance of Downloader has already updated from the API once,
+        # don't do it again. Almost certainly nothing has changed!
+        # Not tested, since CI never waits >5 min with the same Downloader
+        # instantiated.
+        if self._updated_blob_file_from_repo:  # coverage: ignore
+            return None
+
+        reply = self._http_request(self._API_BASE_URL)
+
+        # Extract the SHA hash and the download URL from the response
+
+        # Extract contents to JSON
+        # Not tested, since any URL on the gituhb API that doesn't raise a 404
+        # should return a JSON
+        try:  # coverage: ignore
+            info = reply.json()
+        except requests.exceptions.JSONDecodeError as err:  # coverage: ignore
+            warnings.warn(
+                "URL did not return the expected JSON file: "
+                f"{self._API_BASE_URL}. "
+                f"Response content: {reply.content}. Exception: {err}"
             )
+            self._validate = False
+            return None
+
+        for item in info:
+            try:
+                filename = item["name"]
+                repo_sha = item["sha"]
+                download_url = item["download_url"]
+
+            # Not tested, since any URL on the gituhb API that doesn't return a 404
+            # should be a JSON with these keys
+            except (KeyError, TypeError) as err:  # coverage: ignore
+                warnings.warn(
+                    f"URL {self._API_BASE_URL} returned JSON file, "
+                    "missing expected keys 'sha' and 'download_url`."
+                    f" JSON contents: {info}. Exception: {err}"
+                )
+                filename = None
+                repo_sha = None
+                download_url = None
+
+            if filename is not None:
+                self._update_blob_entry(
+                    filename, repo_sha=repo_sha, download_url=download_url
+                )
+
+        # Save the current epoch time in the blob file as a record of when
+        # it was updated
+        self._blob_dict["_timestamp"] = time.time()
+
+        # At the end, write back to the blobfile
+        self._write_blobfile()
+
+        # The blob file has been updated: set this flag so we won't do it
+        # again on this instance of Downloader
+        self._updated_blob_file_from_repo = True
+
+    def _update_blob_entry(
+        self,
+        filename: str,
```

Comment:
```suggestion
        filename: str | Path,
```
I don't know if this is consistent with elsewhere, though.

Reply:
Actually here it does have to be a string (just the file basename, not the path)